storagemarket/direct_deals_provider.go

```diff
@@ -566,3 +586,50 @@ func (ddp *DirectDealsProvider) FailPausedDeal(ctx context.Context, id uuid.UUID
 
 	return nil
 }
+
+func (ddp *DirectDealsProvider) indexAndAnnounce(ctx context.Context, entry *smtypes.DirectDeal) *dealMakingError {
+	// add deal to piece metadata store
+	ddp.dealLogger.Infow(entry.ID, "about to add direct deal for piece in LID")
+	if err := ddp.piecedirectory.AddDealForPiece(ctx, entry.PieceCID, model.DealInfo{
+		DealUuid:     entry.ID.String(),
+		ChainDealID:  abi.DealID(entry.AllocationID), // Convert the type to avoid migration as underlying types are same
+		MinerAddr:    entry.Provider,
+		SectorID:     entry.SectorID,
+		PieceOffset:  entry.Offset,
+		PieceLength:  entry.Length,
+		CarLength:    uint64(entry.InboundFileSize),
+		IsDirectDeal: true,
+	}); err != nil {
+		return &dealMakingError{
+			retry: types.DealRetryAuto,
+			error: fmt.Errorf("failed to add deal to piece metadata store: %w", err),
+		}
+	}
+	ddp.dealLogger.Infow(entry.ID, "direct deal successfully added to LID")
+
+	// if the index provider is enabled
+	if ddp.ip.Enabled() {
+		if entry.AnnounceToIPNI {
+			// announce to the network indexer but do not fail the deal if the announcement fails,
+			// just retry the next time boost restarts
+			annCid, err := ddp.ip.AnnounceBoostDirectDeal(ctx, entry)
+			if err != nil {
+				return &dealMakingError{
+					retry: types.DealRetryAuto,
+					error: fmt.Errorf("failed to announce deal to network indexer: %w", err),
+				}
+			}
+			ddp.dealLogger.Infow(entry.ID, "announced the direct deal to network indexer", "announcement-cid", annCid)
+		} else {
+			ddp.dealLogger.Infow(entry.ID, "didn't announce the direct deal as requested in the deal proposal")
```

Comment:
```suggestion
			ddp.dealLogger.Infow(entry.ID, "didn't announce the direct deal as requested in the deal proposal because announcements are disabled")
```

Reply:
There is a separate line for when announcements are disabled as index-provider is disabled. This case is for when client requests not to announce

Reply:
Ah ok thanks, I got confused by the language.
Let's change it to
```didn't announce the direct deal because the client disabled announcements for this deal```